qubes/devices.py

```diff
@@ -277,25 +282,27 @@ async def update_required(self, device: Device, required: bool):
 
         # be careful to use already present assignment, not the provided one
         # - to not change options as a side effect
-        if assignment.required == required:
+        if assignment.mode == mode:
             return
 
```

Comment:
Should this refuse mode `manual`? Or maybe automatically convert to unassign call?